src/osm/mod.rs

```diff
@@ -144,45 +144,35 @@ impl OSMTypes {
-                _ => ()
+            if let Some(diffid) = ids.get(&n.id.unwrap()) {
+                diffres.push_str(&*format!(r#"<node old_id="{}" new_id="{}" new_version="{}"/>"#, diffid.old.unwrap(), diffid.new.unwrap(), diffid.version.unwrap()));
             }
         } else if n.action == Some(Action::Modify) {
-            match ids.get(&n.id.unwrap()) {
-                Some(diffid) => { diffres.push_str(&*format!(r#"<node old_id="{}" new_id="{}" new_version="{}"/>"#, n.id.unwrap(), n.id.unwrap(), diffid.version.unwrap())); },
-                _ => ()
-            }
-        } else if n.action == Some(Action::Delete) {
-            match ids.get(&n.id.unwrap()) {
-                Some(_) => { diffres.push_str(&*format!(r#"<node old_id="{}"/>"#, n.id.unwrap())); },
-                _ => ()
+            if let Some(diffid) = ids.get(&n.id.unwrap()) {
+                diffres.push_str(&*format!(r#"<node old_id="{}" new_id="{}" new_version="{}"/>"#, n.id.unwrap(), n.id.unwrap(), diffid.version.unwrap()));
             }
+        } else if n.action == Some(Action::Delete) && ids.get(&n.id.unwrap()).is_some() {
+            diffres.push_str(&*format!(r#"<node old_id="{}"/>"#, n.id.unwrap()));
         }
     }
 
-    for (_i, w) in tree.get_ways() {
+    for w in tree.get_ways().values() {
         if w.action == Some(Action::Create) {
-            match ids.get(&w.id.unwrap()) {
-                Some(diffid) => { diffres.push_str(&*format!(r#"<way old_id="{}" new_id="{}" new_version="{}"/>"#, diffid.old.unwrap(), diffid.new.unwrap(), diffid.version.unwrap())); },
-                _ => ()
+            if let Some(diffid) = ids.get(&w.id.unwrap()) {
+                diffres.push_str(&*format!(r#"<way old_id="{}" new_id="{}" new_version="{}"/>"#, diffid.old.unwrap(), diffid.new.unwrap(), diffid.version.unwrap()));
             }
         } else if w.action == Some(Action::Modify) {
-            match ids.get(&w.id.unwrap()) {
-                Some(diffid) => { diffres.push_str(&*format!(r#"<way old_id="{}" new_id="{}" new_version="{}"/>"#, w.id.unwrap(), w.id.unwrap(), diffid.version.unwrap())); },
-                _ => ()
-            }
-        } else if w.action == Some(Action::Delete) {
-            match ids.get(&w.id.unwrap()) {
-                Some(_) => { diffres.push_str(&*format!(r#"<way old_id="{}"/>"#, w.id.unwrap())); },
-                _ => ()
+            if let Some(diffid) = ids.get(&w.id.unwrap()) {
+                diffres.push_str(&*format!(r#"<way old_id="{}" new_id="{}" new_version="{}"/>"#, w.id.unwrap(), w.id.unwrap(), diffid.version.unwrap()));
             }
+        } else if w.action == Some(Action::Delete) && ids.get(&w.id.unwrap()).is_some() {
+            diffres.push_str(&*format!(r#"<way old_id="{}"/>"#, w.id.unwrap()));
         }
     }
 
-    for (_i, _r) in tree.get_rels() {
+    for _ in tree.get_rels().values() {
```

Comment:
I know this isn't related to this PR, but what's going on here?

Reply:
@mattciferri couldn't tell you without a deep dive into this code. It's been 2 years since I've touched any of this. This part specifically is just a placeholder however. We don't support anything other than points in the OSM upload API at the moment.